### PR TITLE
Support some conversations api (2)

### PR DESCRIPTION
**Pull request by suzuki-shunsuke**
* https://api.slack.com/methods/conversations.invite
* https://api.slack.com/methods/conversations.kick
* https://api.slack.com/methods/conversations.close
* https://api.slack.com/methods/conversations.create
* https://api.slack.com/methods/conversations.info
* https://api.slack.com/methods/conversations.leave
* https://api.slack.com/methods/conversations.replies
* https://api.slack.com/methods/conversations.list
* https://api.slack.com/methods/conversations.open
* https://api.slack.com/methods/conversations.join
* https://api.slack.com/methods/conversations.history

### CODE REVIEW

**Comment by james-lawrence** [comment]
@suzuki-shunsuke i'll look at this saturday morning. initial impressions look good though.

File: conversation.go [hunk 1]
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
 	}
 	return response.Channel, nil
 }
+
+// InviteUsersToConversation invites users to a channel
+func (api *Client) InviteUsersToConversation(channelID string, users []string) (*Channel, error) {
```
Comment (james-lawrence) [inline conversation.go]: thoughts on changing to `InviteUsersToConversation(channelID string, users ...string)`

File: conversation.go [hunk 2]
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
...
+		return err
+	}
+	if !response.Ok {
+		return errors.New(response.Error)
+	}
+	return nil
+}
+
+// CloseConversation closes a direct message or multi-person direct message
+func (api *Client) CloseConversation(channelID string) (bool, bool, error) {
```
Comment (james-lawrence) [inline conversation.go]: returns values could use some documentation. consider making then named. `(noop, alreadyClosed bool, err error)`

File: conversation.go [hunk 3]
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
...
+	Timestamp string
+	Cursor    string
+	Inclusive bool
+	Latest    string
+	Limit     int
+	Oldest    string
+}
+
+// GetConversationReplies retrieves a thread of messages posted to a conversation
+func (api *Client) GetConversationReplies(params *GetConversationRepliesParameters) ([]Message, bool, string, error) {
```
Comment (james-lawrence) [inline conversation.go]: again some documentation over what the return values mean.

File: conversation.go [hunk 4]
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
...
+	Types           []string
+}
+
+// GetConversations returns the list of channels in a Slack team
+func (api *Client) GetConversations(params *GetConversationsParameters) ([]Channel, string, error) {
+	return api.GetConversationsContext(context.Background(), params)
+}
+
+// GetConversationsContext returns the list of channels in a Slack team with a custom context
+func (api *Client) GetConversationsContext(ctx context.Context, params *GetConversationsParameters) ([]Channel, string, error) {
```
Comment (james-lawrence) [inline conversation.go]: return documentation. =)

**Review by james-lawrence — COMMENTED** [review]
minor suggestions, will merge once they are addressed.

**Comment by suzuki-shunsuke** [comment]
@james-lawrence 
Thank you for reviewing.
I have fixed.

**Comment by james-lawrence** [comment]
@suzuki-shunsuke thanks!

**Pull request merged**